packages/components/Button/src/Button.tsx

```diff
@@ -70,17 +94,46 @@ export const Button = compose<ButtonType>({
         });
       }
       const label = accessibilityLabel ?? childText;
-
-      return (
-        <Slots.root {...mergedProps} accessibilityLabel={label}>
+      const buttonContent = (
+        <React.Fragment>
           {loading && <ActivityIndicator />}
           {shouldShowIcon && iconPosition === 'before' && <Slots.icon {...iconProps} />}
           {React.Children.map(children, (child) =>
             typeof child === 'string' ? <Slots.content key="content">{child}</Slots.content> : child,
           )}
           {shouldShowIcon && iconPosition === 'after' && <Slots.icon {...iconProps} />}
-        </Slots.root>
+        </React.Fragment>
       );
+
+      const hasRipple = Platform.OS === 'android';
+      if (hasRipple) {
+        const [extractedMargin, styleWithoutMargin] = extractMarginAndroid(mergedProps.style);
+        return (
+          <Slots.root style={extractedMargin}>
```

Comment:
Can you add a comment here saying Pressable needs to be wrapped in a View to handle rounded corners?